core/platform/email/mailgun_email_services_test.py

```diff
@@ -53,3 +58,129 @@ def test_sending_email(self):
+                    'text': 'body', 'html': 'html'}
+
+        # Lambda function, will replace requests.post() in send_mail.
+        req_post_lambda = (lambda domain_name, auth=None, data=None:
+                           self.assertDictContainsSubset(expected, data))
+        modified_post_request = (self.swap(requests, 'post', req_post_lambda))
+
+        with mailgun_api, mailgun_domain, modified_post_request, allow_emailing:
+            mailgun_email_services.send_mail(
+                feconf.SYSTEM_EMAIL_ADDRESS, feconf.ADMIN_EMAIL_ADDRESS,
+                'subject', 'body', 'html', bcc_admin=False)
+
+
+    def test_bcc_admin_flag(self):
+        """Verifies that the bcc admin flag is working properly in send_mail.
+
+        Note that we replace the requests.post() function in send_mail with
+        an alternate lambda that asserts the correct values were placed in
+        the data dictionary that is then passed to the mailgun api.
+        """
+        mailgun_api = self.swap(feconf, 'MAILGUN_API_KEY', 'api')
+        mailgun_domain = self.swap(feconf, 'MAILGUN_DOMAIN_NAME', 'domain')
+        allow_emailing = self.swap(feconf, 'CAN_SEND_EMAILS', True)
+
+        # Lambda function, will replace requests.post() in send_mail.
+        req_post_lambda = (lambda domain_name, auth=None, data=None:
+                           self.assertEqual(
+                               data['bcc'], feconf.ADMIN_EMAIL_ADDRESS))
+        modified_post_request = self.swap(requests, 'post', req_post_lambda)
+
+        with mailgun_api, mailgun_domain, modified_post_request, allow_emailing:
+            mailgun_email_services.send_mail(
+                feconf.SYSTEM_EMAIL_ADDRESS, feconf.ADMIN_EMAIL_ADDRESS,
+                'subject', 'body', 'html', bcc_admin=True)
+
+
+    def test_reply_to_id_flag(self):
+        """Verifies that the reply_to_id flag is working properly."""
+        mailgun_api = self.swap(feconf, 'MAILGUN_API_KEY', 'api')
+        mailgun_domain = self.swap(feconf, 'MAILGUN_DOMAIN_NAME', 'domain')
+        allow_emailing = self.swap(feconf, 'CAN_SEND_EMAILS', True)
+        reply_id = 123
+
+        # Lambda function, will replace requests.post() in send_mail.
+        req_post_lambda = (lambda domain_name, auth=None, data=None:
+                           self.assertEqual(data['h:Reply-To'],
+                                            'reply+' + str(reply_id) + '@' +
+                                            feconf.INCOMING_EMAILS_DOMAIN_NAME))
+        modified_post_request = (self.swap(requests, 'post',
+                                           req_post_lambda))
+
+        with mailgun_api, mailgun_domain, modified_post_request, allow_emailing:
+            mailgun_email_services.send_mail(
+                feconf.SYSTEM_EMAIL_ADDRESS, feconf.ADMIN_EMAIL_ADDRESS,
+                'subject', 'body', 'html',
+                bcc_admin=False, reply_to_id=reply_id)
+
+
```

Comment:
Extra newline

Reply:
All of the lines you listed as "extra newline" failed pylint by exactly one character :( 
(81/80 i think)

Reply:
I don't understand... There needs to be one newline between functions right? How are there 81 characters?

If pylint failed, it would probably have said something like expected 2 newlines, found 1 or something like that.

Reply:
Ah, I thought you meant the line with code was incorrectly split among two lines. I'll fix that.